docs/maintainers/release.md

```diff
@@ -1,44 +1,78 @@
-  2. a commit to update [VERSION](../../VERSION) as needed.
-
-* Make the release on Github with the release branch as the target: copy the
-  relevant section of the [CHANGELOG](../../CHANGELOG) for the release
-  description and check the `pre-release` box if applicable. There is no need to
-  upload any assets as this will be done automatically by a Github workflow,
-  after you create the release.
-
-* After a while (time for the Github workflows to complete), check that:
-  1. the docker image has been pushed to
+release. We use `<TAG>` as a placeholder for the release tag (e.g. `v1.4.0`).
+
+1. *For a minor release* On the code freeze date (typically one week before the
+   actual scheduled release date), create a release branch for the new minor
+   release (e.g `release-1.4`).
+   - after that time, only bug fixes should be merged into the release branch,
+     by [cherry-picking](../contributors/cherry-picks.md) the fix after it has
+     been merged into main. The maintainer in charge of that specific minor
+     release can either do the cherry-picking directly or ask the person who
+     contributed the fix to do it.
+
+2. Open a PR against the appropriate release branch with the following commits:
+   - a commit to update the [CHANGELOG](../../CHANGELOG). *For a minor release*,
+     all significant changes and all bug fixes since the first version of the
+     previous minor release should be mentioned, even bug fixes which have
+     already been included in some patch release. *For a patch release*, you
+     will mention all the bug fixes since the previous release with the same
+     minor version. The commit message must be *exactly* `"Update CHANGELOG for
+     <TAG> release"`, as a bot will look for this commit and cherry-pick it to
+     update the main branch (starting with Antrea v1.0). The
+     [process-changelog.go](../../hack/release/process-changelog.go) script may
+     be used to easily generate links to PRs and the Github profiles of PR
+     authors.
+   - a commit to update [VERSION](../../VERSION) as needed, using the following
+     commit message: `"Set VERSION to <TAG>"`.
+
+3. Run all the tests for the PR, investigating test failures and re-triggering
+   the tests as needed.
+   - Github worfklows are run automatically whenever the head branch is updated.
+   - Jenkins tests need to be [triggered manually](../../CONTRIBUTING.md#getting-your-pr-verified-by-ci).
+   - Cloud tests need to be triggered manually through the
+     [Jenkins web UI](https://jenkins.antrea-ci.rocks/). Admin access is
+     required. For each job (AKS, EKS, GKE), click on `Build with Parameters`,
+     and enter the name of your fork as `ANTREA_REPO` and the name of your
+     branch as `ANTREA_GIT_REVISION`. Test starting times need to be staggered:
+     if multiple jobs run at the same time, the Jenkins worker may run
+     out-of-memory.
+
+4. Request a review from the other maintainers, and anyone else who may need to
```

Comment:
side note: with the new process, shall we add a new "release" label which automatically would request review from maintainers?

Reply:
done, thanks for the suggestion